### PR TITLE
feat: add warning message if store id already exists (#1394)

### PR DESCRIPTION
**Pull request by zavsievich**
Add a toast message if store with current id already exists.

### CODE REVIEW

**Comment by posva** [comment]
Thanks but this doesn't do what you think it does 😓 , it doesn't add the feature you linked. Did you try it?

**Comment by zavsievich** [comment]
> Thanks but this doesn't do what you think it does 😓 

Could you please explain how to fix this?

File: packages/pinia/src/store.ts [hunk 1]
```diff
@@ -900,6 +902,11 @@ export function defineStore(
         // @ts-expect-error: not the right inferred type
         useStore._pinia = pinia
       }
+    } else {
+      toastMessage(
+        `Store with id: ${id} already exists. Stores shoud have unique id.`,
```
Comment (BlackCrowxyz) [inline packages/pinia/src/store.ts]:
```suggestion
        `Store with id: ${id} already exists. Stores should have unique id.`,
```

**Comment by BlackCrowxyz** [comment]
> > Thanks but this doesn't do what you think it does 😓
> 
> Could you please explain how to fix this?

When changing routes in playground, you can see unnecessary warning messages. Pinia registers stores with `id` when creating them, reusing stores in other places mean they were registered before with the `id` so the warning will come up!

**Review by OXXXRA — APPROVED** [review]

**Review by OXXXRA — APPROVED** [review]

**Comment by posva** [comment]
Thanks but this does not implement the linked feature, it warns whenever the store is reused (which we don't want ofc).

We don't know if we will implement the feature yet but if you want to give it another shot make sure to:

- include tests
- make the warning dev only (just a warn, not the toast)

**Pull request closed without merging**